Submitted By:     VMiklos <mamajom@axelero.hu>
Date:             2005.01.21
Initial Packages: 2.9.5
Description:      This patch fixes an annoying bug in pacman.
                  If a package has an install script, and pacman has a suid bit,
                  now running these script will be possible.

diff -Naur pacman-2.9.5.orig/src/pacman.c pacman-2.9.5/src/pacman.c
--- pacman-2.9.5.orig/src/pacman.c	2005-01-12 00:14:16.000000000 +0100
+++ pacman-2.9.5/src/pacman.c	2005-01-21 23:44:18.000000000 +0100
@@ -3326,10 +3326,10 @@
 
 	vprint("Executing %s script...\n", script);
 	if(oldver) {
-		snprintf(cmdline, PATH_MAX, "echo \"umask 0022; source %s %s %s %s\" | chroot %s /bin/sh",
+		snprintf(cmdline, PATH_MAX, "echo \"umask 0022; source %s %s %s %s\" | /usr/sbin/chroot %s /bin/sh",
 				scriptpath, script, ver, oldver, pmo_root);
 	} else {
-		snprintf(cmdline, PATH_MAX, "echo \"umask 0022; source %s %s %s\" | chroot %s /bin/sh",
+		snprintf(cmdline, PATH_MAX, "echo \"umask 0022; source %s %s %s\" | /usr/sbin/chroot %s /bin/sh",
 				scriptpath, script, ver, pmo_root);
 	}
 	vprint("%s\n", cmdline);
